clients/venice-client/src/main/java/com/linkedin/venice/fastclient/ClientConfig.java

```diff
@@ -108,13 +115,19 @@ private ClientConfig(
       StoreMetadataFetchMode storeMetadataFetchMode,
       D2Client d2Client,
       String clusterDiscoveryD2Service,
-      boolean useStreamingBatchGetAsDefault) {
+      boolean useStreamingBatchGetAsDefault,
+      boolean useGrpc,
+      Map<String, String> nettyServerToGrpcAddressMap) {
     if (storeName == null || storeName.isEmpty()) {
       throw new VeniceClientException("storeName param shouldn't be empty");
     }
     if (r2Client == null) {
       throw new VeniceClientException("r2Client param shouldn't be null");
     }
+    if (useGrpc && nettyServerToGrpcAddressMap == null) {
```

Comment:
BTW, with such step, it won't work in EI/Prod env, which means we can't test/benchmark it in EI env.
Since leveraging Helix Instance config can be a more involved change, can we create a new endpoint in Netty Server, so that when grpc fast-client discovers the netty address via metadata request, it can send an request to the Netty server to figure out the GRPC port and cache it locally.
This may not be the ultimate solution, but it can work with minimal overhead since cluster setup doesn't change often.

If you agree, you can build such mechanism in a following PR.